src/NuGet.CommandLine/MsBuildUtility.cs

```diff
@@ -119,46 +119,13 @@ public static string GetMsbuildDirectory(string version)
         {
             if (string.IsNullOrEmpty(version))
             {
-                // default to 4.0
-                version = "4.0";
-
-                // run msbuild to get the version
-                var processStartInfo = new ProcessStartInfo
-                {
-                    UseShellExecute = false,
-                    FileName = "msbuild.exe",
-                    Arguments = "/version",
-                    RedirectStandardOutput = true
-                };
-
-                try
-                {
-                    using (var process = Process.Start(processStartInfo))
-                    {
-                        process.WaitForExit(10 * 1000);
-
-                        if (process.ExitCode == 0)
-                        {
-                            var output = process.StandardOutput.ReadToEnd();
-
-                            // The output of msbuid /version with MSBuild 14 is:
-                            //
-                            // Microsoft (R) Build Engine version 14.0.23107.0
-                            // Copyright(C) Microsoft Corporation. All rights reserved.
-                            //
-                            // 14.0.23107.0                            
-                            var lines = output.Split(
-                                new[] { Environment.NewLine },
-                                StringSplitOptions.RemoveEmptyEntries);
-
-                            version = lines.LastOrDefault(
-                                line => !string.IsNullOrWhiteSpace(line));
-                        }
-                    }
-                }
-                catch
+                // use the highest installed version
+                using (var projectCollection = new ProjectCollection())
                 {
-                    // ignore errors
+                    var highestVersionToolset = projectCollection.Toolsets
+                        .OrderByDescending(toolset => new Version(toolset.ToolsVersion))
+                        .First();
```

Comment:
If .net framework is installed, then at least one msbuild is also installed. So no need to check for empty list here.